extensions/config-yaml/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/YamlConfigSourceProvider.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_JAR_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_JAR_PRIORITY;
+
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+public class YamlConfigSourceProvider implements ConfigSourceProvider {
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader forClassLoader) {
+        ClassLoader cl = getClassLoader();
+        return Arrays.asList(
+                new YamlConfigSource(cl.getResource(MICROPROFILE_CONFIG_YML_FILE), MICROPROFILE_CONFIG_YML_JAR_PRIORITY),
+                new YamlConfigSource(cl.getResource(APPLICATION_YML_FILE), APPLICATION_YML_JAR_PRIORITY),
+                new YamlConfigSource(getFileSystemURL(MICROPROFILE_CONFIG_YML_FILE),
```

Comment:
Perhaps check if the URL is not null before adding to the result list?

Reply:
Done.